### PR TITLE
added metrics, migrated to Grafana Agent flow

### PR DESCRIPTION
**Pull request by taco-paco**

### CODE REVIEW

File: api/src/handlers/utils.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            if val.is_successful() {
+                metrics
+                    .action_successes_total
+                    .with_label_values(&[action_label_value])
+                    .inc();
+                Ok(val)
+            } else {
+                metrics
+                    .action_failures_total
+                    .with_label_values(&[action_label_value]);
```
Comment (varex83) [inline api/src/handlers/utils.rs:40]: you forgot to increment the metric here

File: api/src/handlers/utils.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                Ok(val)
+            }
+        }
+        Err(err) => {
+            metrics
+                .action_failures_total
+                .with_label_values(&[action_label_value]);
```
Comment (varex83) [inline api/src/handlers/utils.rs:47]: ...and here

File: api/src/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+}
+
+impl Metrics {
+    fn update_metrics(&self, req: &mut Request<'_>) {
+        match req.uri().path().as_str() {
+            "/compile-scarb"
+            | "/compile-scarb-async"
+            | "/compile-to-sierra"
+            | "compile-to-sierra-async" => self
```
Comment (varex83) [inline api/src/metrics.rs:58]:
```suggestion
            | "/compile-to-sierra-async" => self
```

File: api/src/handlers/types.rs [hunk 4]
```diff
@@ -23,6 +33,12 @@ pub struct ScarbCompileResponse {
     pub file_content_map_array: Vec<FileContentMap>,
 }
 
+impl Successable for ScarbCompileResponse {
+    fn is_successful(&self) -> bool {
+        self.message == "Success"
```
Comment (varex83) [inline api/src/handlers/types.rs:38]:
```suggestion
        self.message.to_lower_case() == "success"
```

Comment (varex83) [inline api/src/handlers/types.rs:38]: Should it check status? Can we make status enum?

File: api/src/handlers/utils.rs (hunk 1 above)
Comment (taco-paco) [reply api/src/handlers/utils.rs:40]: Good catch, thanks!

File: api/src/handlers/utils.rs (hunk 2 above)
Comment (taco-paco) [reply api/src/handlers/utils.rs:47]: fixed

File: api/src/metrics.rs (hunk 3 above)
Comment (taco-paco) [reply api/src/metrics.rs:58]: Fixed

File: api/src/handlers/types.rs (hunk 4 above)
Comment (taco-paco) [reply api/src/handlers/types.rs:38]: Well, our responses need to be changed in general to follow restful practices. This was done in horizontal scaling work on zksync side, but it's abandoned now. Anyway, it is a whole separate issue

Comment (taco-paco) [reply api/src/handlers/types.rs:38]: Well, our responses need to be changed in general to follow restful practices. This was done in horizontal scaling work on zksync side, but it's abandoned now. Anyway, it is a whole separate issue

Comment (taco-paco) [reply api/src/handlers/types.rs:38]: Seems redundant, better to standardize responses

**Review by varex83 — APPROVED** [review]
LGTM

**Pull request merged**